types/redux-form/lib/reducer.d.ts

```diff
@@ -18,7 +18,7 @@ export interface FormStateMap {
 }
 
 export interface FormState {
-    registeredFields: RegisteredFieldState[];
+    registeredFields: { [name: string]: RegisteredFieldState };
```

Comment:
From what I can tell, the state in the store is read/written based on calls to [`getIn`](https://github.com/redux-form/redux-form/blob/0c4a04777d812b4f6769bbcb26e3d7eef359a586/src/structure/plain/getIn.js#L4) and [`setIn`](https://github.com/redux-form/redux-form/blob/0c4a04777d812b4f6769bbcb26e3d7eef359a586/src/structure/plain/setIn.js#L39), which theoretically allow the state to be either an object or an array?

The test here shows `registeredFields` being used as an array: https://github.com/redux-form/redux-form/blob/0485967d7f98b349e850570a2f2f3a4124f392d8/src/__tests__/helpers/reducer.reset.js#L135

I honestly don't know enough about `redux-form` to know whether this is correct.